app/questionnaire/questionnaire_schema.py

```diff
@@ -91,6 +105,48 @@ def when_rules_section_dependencies_by_section(
     ) -> ImmutableDict[str, set[str]]:
         return ImmutableDict(self._when_rules_section_dependencies_by_section)
 
+    @cached_property
+    def when_rules_section_dependencies_for_progress(
+        self,
+    ) -> defaultdict[str, set[str]]:
+        when_rules_section_dependencies_for_progress = defaultdict(set)
+        for (
+            section,
+            dependent,
+        ) in (
+            self._when_rules_block_dependencies_by_section_for_progress_value_source.items()
+        ):
+            section_dependents = get_flattened_mapping_values(dependent)
+            for dependent_section in section_dependents:
+                when_rules_section_dependencies_for_progress[dependent_section] = {
+                    section
+                }
+        for (
+            section,
+            dependents,
+        ) in (
+            self._when_rules_section_dependencies_by_section_for_progress_value_source.items()
+        ):
+            for dependent_section in dependents:
+                when_rules_section_dependencies_for_progress[dependent_section] = {
+                    section
+                }
```

Comment:
At the moment this doesn't account for multiple dependents, it is being set to a single set, use `.add` instead. This needs to be updated to handle that and schema/unit test updated to reflect this as no test broke. If you try this on calculated summary extended, you can see the mapping is incorrect. A unit test which loads that schema and asserts these this property would be enough.

Reply:
Updated and added test 👍

Reply:
Since it is a complicated test, it is worth adding some comments on what is happening.

Reply:
Added a comment 👍

Reply:
I left it on the wrong one, useful here to. But I originally meant against `test_questionnaire_store_updater_dependency_capture`. Test name should be updated, maybe `test_questionnaire_store_updater_dependency_revaluation_order`

Reply:
Added a comment for this test 👍